jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
> There is a check [here](https://github.com/javier-gomez-hs/killbill/blob/e7c1e41dc2492843cbf5af7c57972f3b6fc053ef/payment/src/main/java/org/killbill/billing/payment/api/DefaultPaymentApi.java#L295-L295) to verify that the transactionExternalKey is not null.

That's odd and it doesn't look consistent. Let me check with @sbrossie and I'll get back to you. Good catch! 💯 

Probably related: https://github.com/killbill/killbill/issues/349.